xml/art_sles_amd-sev.xml

```diff
@@ -0,0 +1,431 @@
+   owner's base64-encoded Diffie-Hellman (DH) key. The key is used to negotiate
+   a master secret key between the SEV firmware and guest owner. This master
+   secret key is then used to establish a trusted channel between SEV firmware
+   and guest owner.
+  </para>
+  <para>
+   The optional <literal>&lt;session&gt;</literal> element provides the guest
+   owner's base64-encoded session blob, as defined in the SEV API spec. See the
+   LAUNCH_START section of the SEV specification for the session-blob format.
+  </para>
+  <para>
+   In addition to the <literal>&lt;launchSecurity&gt;</literal> settings,
+   SEV-encrypted VMs must have the <literal>iommu='on'</literal> attribute set
+   in each <literal>virtio</literal> device, as shown in the example
+   configuration. This attribute is required in order to enable DMA APIs for
+   the device within QEMU.
+  </para>
+  <para>
+   SEV-encrypted VMs must also have all of their memory regions locked to allow
+   DMA and prevent swapping. This includes not only the VM's RAM, but also
+   ROM(s), pflash, and video memory. Using the example configuration, the memory
+   regions needing to be locked total to 4352592 kB, which exceeds the default
+   <literal>memlock</literal> limit. The <literal>memlock</literal> limit for a
+   virtual machine process can be increased using the <literal>&lt;hard_limit&gt;</literal>
+   subelement of <literal>&lt;memtune&gt;</literal>. The value 4352592 kB was
+   determined using the following formula:
+  </para>
+  <screen>hard_limit = VM RAM + VM video memory + ROMs/ACPI tables</screen>
+  <para>
+   where:
+  </para>
+<screen>    4096 kB (UEFI ROM)
+ +  4096 kB (UEFI variable store)
+ +   128 kB (PC ROM)
+ +   128 kB (ISA BIOS) 
+ +  2384 kB (ACPI tables)
+ = 10832 kB (total of all ROMs/ACPI tables)</screen>
+  <para>
+   Although 10832 kB for ROMs and ACPI tables should be sufficient in most
+   cases, it is recommended to add a few hundred kilobytes of padding. ROM sizes
+   have been known to change and adding some padding here will accommodate that.
+   Using the virtual machine RAM and video memory values from the example
+   configuration:
+  </para>
+<screen>  4194304 kB (virtual machine RAM) 
++   65536 kB (qlx RAM) 
++   65536 kB (qlx video RAM) 
++   16384 kB (qlx VGA memory)
++   10832 kB (ROMs/ACPI tables)
+= 4352592 kB hard_limit</screen>
+  <para>
+   Calculating the <literal>memlock</literal> limit and setting it via 
+   <literal>&lt;hard_limit&gt;</literal> can be avoided by configuring the
+   virtual machine to use <literal>hugepages</literal>. For more information on
+   using hugepages with VMs, refer to the <citetitle>Virtualization Best Practices Guide</citetitle>, 
+   Chapter "Configuring the VM Host Server and the VM Guest to use Huge Pages":
+   <link xlink:href="https://www.suse.com/documentation/sles-15/book_quickstarts/data/sec_vt_best_hostlevel.html#sec_vt_best_mem_huge_pages"/>.
```

Comment:
IMHO, this external link is a bit dangerous. :wink:  You can avoid this external link if you integrate your file into  `book_quickstarts.xml` and use `<xref/>`. The above sentence would look like this:

```xml
For more information on using hugepages with VMs, refer to <xref linkend="sec.vt.best.mem.huge_pages"/>.
```

This would generate the above link automatically with all the necessary book and chapter titles.

Reply:
I chose this linking method following some fairly extensive discussion, of which you were a part!
As I understand it, this is to be a standalone article, published as (for example) a PDF on AMD's website and in other places. Would making it a sub-chapter of another book cause problems with this?